libraries/matrix/src/main/kotlin/io/element/android/x/matrix/FakeMatrixClient.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.x.matrix
+
+import io.element.android.x.matrix.core.RoomId
+import io.element.android.x.matrix.core.SessionId
+import io.element.android.x.matrix.core.UserId
+import io.element.android.x.matrix.media.FakeMediaResolver
+import io.element.android.x.matrix.media.MediaResolver
+import io.element.android.x.matrix.room.FakeMatrixRoom
+import io.element.android.x.matrix.room.InMemoryRoomSummaryDataSource
+import io.element.android.x.matrix.room.MatrixRoom
+import io.element.android.x.matrix.room.RoomSummaryDataSource
+import org.matrix.rustcomponents.sdk.MediaSource
+
+class FakeMatrixClient(override val sessionId: SessionId) : MatrixClient {
```

Comment:
This class must be in the `test` folder.

Reply:
We can't put in test folder as it'll be available only internally. These classes are used on other modules, so we have to access them. But what we can do is create a matrix-test module

Reply:
OK. Maybe use this pattern to fix that?
```
dependencies {
  testCompile project(':libraries:matrix').sourceSets.test.output
}
```

Reply:
In an ideal world we'd use textFixtures gradle feature, but it's not working over AGP
Adding a module is still cleaner than relying on sourceSets.test, otherwise we might end up with UnitTests from matrix module